File: python/leetcode/435_Non-overlapping_Intervals.py
# /usr/bin/python3.6
# -*- coding:utf-8 -*-


class Solution(object):
    def eraseOverlapIntervals(self, intervals):
        """
        :type intervals: List[List[int]]
        :rtype: int
        """
        intervals.sort()
        old = [-1<<31, -1<<31]
        ret = 0
        for x, y in intervals:
            if x>= old[1]:
                old = x, y
                ret += 1
            else:
                if y < old[1]:
                    old = x, y
        return len(intervals) - ret


def main():
    s = Solution()


if __name__ == "__main__":
    main()

File: C++/LeetCode/LeetCode/435_Non-overlapping_Intervals.cpp
#include <vector>
#include <algorithm>
using namespace std;
class Solution {
public:
	static bool cmp(vector<int>& a, vector<int>& b) {
		if (a[0] < b[0]) {
			return true;
		}
		else if (a[0] == b[0]) {
			if (a[1] <= b[1]) {
				return true;
			}
			return false;
		}
		return false;
	}
	int eraseOverlapIntervals(vector<vector<int>>& intervals) {
		sort(intervals.begin(), intervals.end(), cmp);
		int ret = 0;
		if (intervals.size() == 0) {
			return 0;
		}
		vector<int> old = intervals[0];
		for (int i = 1; i<intervals.size(); i++) {
			if (intervals[i][0] >= old[1]) {
				old = intervals[i];
			}
			else {
				if (intervals[i][1] < old[1]) {
					old = intervals[i];

				}
				ret++;
			}
		}
		return ret;
	}
};